docs/dev_guide/diagnostic_ref/dpct1042.rst

```diff
@@ -19,85 +19,138 @@ Detailed Help
+    void test_function(int *out) {
+      kernel<<<1, 1>>>(out);
+    }
+
+results in the following migrated SYCL code:
 
 .. code-block:: cpp
    :linenos:
 
-     ... // declarations for device0 to device30
-     dpct::constant_memory<int, 1> device32(ARRAY_SIZE + ARRAY_SIZE); // one buffer instead of two
- 
-     // kernel function declaration
-     void kernel(sycl::nd_item<3> item_ct1, int *device0, ..., int *device32);
- 
-     void test_function() {
-       ... // memcpy operations for device0 to device30
-       dpct::dpct_memcpy(device31.get_ptr(), host, ARRAY_SIZE * sizeof(int));
-       // memory copy with offset:
-       dpct::dpct_memcpy(device31.get_ptr() + ARRAY_SIZE, host,
-                         ARRAY_SIZE * sizeof(int));
-       {
-         dpct::get_default_queue().submit([&](sycl::handler &cgh) {
-           ... // accessors for device0 to device30, captured by SYCL kernel lambda
-               auto device31_acc_ct1 =
-                   device31.get_access(cgh); // only one accessor instead of 2
-           cgh.parallel_for(sycl::nd_range<3>(sycl::range<3>(1), sycl::range<3>(1)),
-                            [=](sycl::nd_item<3> item_ct1) {
-                              // last parameter is modified to use offset and the
-                              // same accessor as previous parameter
-                              kernel(item_ct1, ... /*arguments device0 to device30*/
-                                               device31_acc_ct1.get_pointer(),
-                                     device31_acc_ct1.get_pointer() + ARRAY_SIZE);
-                            });
-         });
-       }
-     }
+    #define ARRAY_SIZE 2
+
+    static dpct::constant_memory<int, 1> device0(ARRAY_SIZE);
+    static dpct::constant_memory<int, 1> device1(ARRAY_SIZE);
+    static dpct::constant_memory<int, 1> device2(ARRAY_SIZE);
+    ...
+    static dpct::constant_memory<int, 1> device30(ARRAY_SIZE);
+    static dpct::constant_memory<int, 1> device31(ARRAY_SIZE);
+    
+    // kernel function declaration
+    void kernel(int *out, const sycl::nd_item<3> &item_ct1, int const *device0,
+                int const *device1, int const *device2, ..., int const *device30,
+                int const *device31) {
+      int i = item_ct1.get_local_range(2) * item_ct1.get_group(2) +
+              item_ct1.get_local_id(2);
+      out[i] = device0[i] + device1[i] + device2[i] +
+               ... +
+               device30[i] + device31[i];
+    }
+    
+    void test_function(int *out) {
+
+      /*
+      DPCT1042:0: The size of the arguments passed to the SYCL kernel exceeds the
+      minimum size limit (1024) for a non-custom SYCL device. You can get the
+      hardware argument size limit by querying info::device::max_parameter_size. You
+      may need to rewrite this code if the size of the arguments exceeds the
+      hardware limit.
+      */
+      dpct::get_out_of_order_queue().submit([&](sycl::handler &cgh) {
+        device0.init();

```

Comment:
In latest migration code, *init() functions should be out of command group.  So please refresh the code.

Reply:
ok